internal/controller/sync.go

```diff
@@ -541,124 +411,116 @@ func (r *Route) buildNextCR(ctx context.Context, route *routev1.Route, revision
 	if metav1.HasAnnotation(route.ObjectMeta, cmapi.SubjectSerialNumberAnnotationKey) {
 		serialNumber = route.Annotations[cmapi.SubjectSerialNumberAnnotationKey]
 	}
-	csr, err := x509.CreateCertificateRequest(
-		rand.Reader,
-		&x509.CertificateRequest{
-			Version:            0,
-			SignatureAlgorithm: signatureAlgorithm,
-			PublicKeyAlgorithm: publicKeyAlgorithm,
-			Subject: pkix.Name{
-				CommonName:         route.Annotations[cmapi.CommonNameAnnotationKey],
-				Country:            countries,
-				Locality:           localities,
-				Organization:       organizations,
-				OrganizationalUnit: organizationalUnits,
-				PostalCode:         postalCodes,
-				Province:           provinces,
-				SerialNumber:       serialNumber,
-				StreetAddress:      streetAddresses,
-			},
-			EmailAddresses: emailAddresses,
-			DNSNames:       dnsNames,
-			IPAddresses:    ipSans,
-			URIs:           uriSans,
-		},
-		key,
-	)
-	if err != nil {
-		return nil, err
-	}
-	csrPEM := pem.EncodeToMemory(&pem.Block{
-		Type:  "CERTIFICATE REQUEST",
-		Bytes: csr,
-	})
-
 	var issuerName string
 	if metav1.HasAnnotation(route.ObjectMeta, cmapi.IngressIssuerNameAnnotationKey) {
 		issuerName = route.Annotations[cmapi.IngressIssuerNameAnnotationKey]
 	} else {
 		issuerName = route.Annotations[cmapi.IssuerNameAnnotationKey]
 	}
 
-	cr := &cmapi.CertificateRequest{
+	secretName := route.Name + "-tls-cert"
+
+	// Build the Certificate resource with the collected information
+	// https://cert-manager.io/docs/usage/certificate/
+	cert := &cmapi.Certificate{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: route.Name + "-",
 			Namespace:    route.Namespace,
-			Annotations:  map[string]string{cmapi.CertificateRequestRevisionAnnotationKey: strconv.Itoa(revision + 1)},
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(
 					route,
 					routev1.GroupVersion.WithKind("Route"),
 				),
 			},
 		},
-		Spec: cmapi.CertificateRequestSpec{
-			Duration: &metav1.Duration{Duration: duration},
+		Spec: cmapi.CertificateSpec{
+			SecretName:     secretName,
+			Duration:       &metav1.Duration{Duration: duration},
+			EmailAddresses: emailAddresses,
+			// RenewBefore?
+			// RevisionHistoryLimit?
+			CommonName: route.Annotations[cmapi.CommonNameAnnotationKey],
+			Subject: &cmapi.X509Subject{
+				Countries:           countries,
+				Localities:          localities,
+				Organizations:       organizations,
+				OrganizationalUnits: organizationalUnits,
+				PostalCodes:         postalCodes,
+				Provinces:           provinces,
+				SerialNumber:        serialNumber,
+				StreetAddresses:     streetAddresses,
+			},
+			PrivateKey: &cmapi.CertificatePrivateKey{
+				Algorithm: privateKeyAlgorithm,
+				Size:      privateKeySize,
+				// RotationPolicy?
+			},
+			DNSNames: dnsNames,
+			// TODO:
```

Comment:
^^ TODO

Reply:
As in the description, the intention is not to merge this as is but to decide whether to go with this or #102 and then proceed with the one we choose!